packages/sanity/src/core/versions/components/BundleMenu.tsx

```diff
@@ -0,0 +1,126 @@
+import {CheckmarkIcon} from '@sanity/icons'
+import {Box, Flex, Menu, MenuButton, MenuDivider, MenuItem, Spinner, Text} from '@sanity/ui'
+import {type ReactElement, useCallback, useContext} from 'react'
+import {RelativeTime} from 'sanity'
+import {styled} from 'styled-components'
+
+import {
+  VersionContext,
+  type VersionContextValue,
+} from '../../../_singletons/core/form/VersionContext'
+import {type BundleDocument} from '../../store/bundles/types'
+import {LATEST} from '../util/const'
+import {isDraftOrPublished} from '../util/dummyGetters'
+import {BundleBadge} from './BundleBadge'
+
+const StyledMenu = styled(Menu)`
+  min-width: 200px;
+`
+
+interface BundleListProps {
+  button: ReactElement
+  bundles: BundleDocument[] | null
+  loading: boolean
+  actions?: ReactElement
+}
+
+export function BundleMenu(props: BundleListProps): JSX.Element {
+  const {bundles, loading, actions, button} = props
+
+  // eslint-disable-next-line no-warning-comments
+  // TODO MAKE SURE THIS IS HOW WE WANT TO DO THIS
+  const {currentVersion, setCurrentVersion, isDraft} =
+    useContext<VersionContextValue>(VersionContext)
+
+  // eslint-disable-next-line no-warning-comments
+  // FIXME REPLACE WHEN WE HAVE REAL DATA
+  const handleBundleChange = useCallback(
+    (bundle: Partial<BundleDocument>) => () => {
+      setCurrentVersion(bundle)
+    },
+    [setCurrentVersion],
+  )
+
+  return (
+    <>
+      <MenuButton
+        button={button}
+        id="global-version-menu"
+        menu={
+          <StyledMenu>
+            {loading ? (
+              <Flex padding={4} justify="center">
+                <Spinner muted />
+              </Flex>
+            ) : (
+              <>
+                <MenuItem
+                  iconRight={isDraft ? <CheckmarkIcon /> : undefined}
+                  onClick={handleBundleChange(LATEST)}
+                  pressed={false}
+                  text={LATEST.title}
+                />
+                {bundles && bundles.length > 0 && (
+                  <>
+                    <MenuDivider />
+                    {bundles
+                      .filter((b) => !isDraftOrPublished(b.name) && !b.archived)
```

Comment:
`b.archived` doesn't exist on `BundleDocument` it seems though. I know we'd discussed elsewhere around having `archivedAt`

Reply:
Yeah I made this change before the archivedAt was discussed 😄 
@juice49 & @pedrobonamin are we ok with making the change to `archivedAt` ?

Reply:
I remeber that Pedro at one point or another mentioned that we might want to make use of "status" instead, not sure where we landed on this

Reply:
Landed on keeping the `archivedAt`!